packages/use-i18n/src/usei18n.tsx

```diff
@@ -99,19 +108,23 @@ interface Context<Locale extends BaseLocale> {
 // useI18n / useTranslation requires to explicitely give a Locale to use.
 const I18nContext = createContext<Context<any> | undefined>(undefined)
 
-export function useI18n<Locale extends BaseLocale>(): Context<Locale> {
+export function useI18n<
+  // eslint-disable-next-line @typescript-eslint/ban-types
+  Locale extends BaseLocale = {},
```

Comment:
Because there is no other way to detect if the user provided a generic or not. We have to set is to an empty object, and if this objects doesn't have any keys (`keyof Locale extends never`) we return the errors.